### ISSUE TITLE
Remove `num_rational` from dependency tree

### ISSUE THREAD

**Issue by hdevalence** [penumbra-zone/penumbra#1466]
We shouldn't be outsourcing consensus-critical arithmetic to one of the `num` crates, but this is currently used in the governance code to do tallies.

**Comment by plaidfinch** [comment]
Is there something in particular about these crates that is worrisome to include in the TCB? Happy to go reimplement rational arithmetic (or vendor `num_rational`?) but curious about the reasoning here.

**Comment by hdevalence** [comment]
Yeah, the worry is that our consensus rules include a third-party implementation of big-integer arithmetic. We don't actually need rational arithmetic, we just need an implementation of a fractional threshold, so it'd be simpler just to hardcode the checks in the governance implementation.
